### ISSUE TITLE
Crash loading video (Android)

### ISSUE THREAD

**Issue by gijoehosaphat** [TheWidlarzGroup/react-native-video#208]
Getting this crash during a users path through our app. It does not happen on any consistent basis. I believe some sort of race-condition is involved.

```
Fatal Exception: java.lang.IllegalStateException
       at android.media.MediaPlayer.prepareAsync(MediaPlayer.java)
       at com.yqritc.scalablevideoview.ScalableVideoView.prepareAsync(ScalableVideoView.java:177)
       at com.brentvatne.react.ReactVideoView.setSrc(ReactVideoView.java:147)
       at com.brentvatne.react.ReactVideoView.onAttachedToWindow(ReactVideoView.java:286)
       at android.view.View.dispatchAttachedToWindow(View.java:13607)
       at android.view.ViewGroup.addViewInner(ViewGroup.java:3981)
       at android.view.ViewGroup.addView(ViewGroup.java:3792)
       at android.view.ViewGroup.addView(ViewGroup.java:3733)
       at com.facebook.react.views.view.ReactViewManager.addView(ReactViewManager.java:196)
       at com.facebook.react.views.view.ReactViewManager.addView(ReactViewManager.java:39)
       at com.facebook.react.uimanager.NativeViewHierarchyManager.manageChildren(NativeViewHierarchyManager.java:366)
       at com.facebook.react.uimanager.UIViewOperationQueue$ManageChildrenOperation.execute(UIViewOperationQueue.java:175)
       at com.facebook.react.uimanager.UIViewOperationQueue$2.run(UIViewOperationQueue.java:740)
       at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:804)
       at com.facebook.react.uimanager.GuardedChoreographerFrameCallback.doFrame(GuardedChoreographerFrameCallback.java:32)
       at com.facebook.react.uimanager.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:125)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:790)
       at android.view.Choreographer.doCallbacks(Choreographer.java:596)
       at android.view.Choreographer.doFrame(Choreographer.java:556)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:778)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:155)
       at android.app.ActivityThread.main(ActivityThread.java:5725)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1030)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:825)
```

As a test I introduced a TimerMixin delay to loading the video in componentDidMount which caused a dramatic decrease in the frequency, but is obviously not a real solution to the crash.

One trait about this crash is that it happens much more often on devices, and rarely if ever in a simulator like Genymotion.

Another note about this is that the videos are local.

**Comment by donnguyen** [comment]
I am running into the same issue here 😞

**Comment by gravitinos** [comment]
I seem to have found a memory leak in unmounting the video component on Android, which I think is related to this, although they might be two separate problems I'm running into at once.

I ran into it running my app overnight, which switched through images and videos every 5-30 seconds. I was able to replicate this within a few hours by changing it to every second it switches.

I avoided it by never unmounting the Video component, but updating it (so I don't rebuild the component, ONLY update it).

Overtime, the CPU slowly builds up. Depending on the device, it would max out and make the app run horribly (but with no crash, maybe leveling out at about 50% CPU).

Running a systrace, I could see the process that was taking up the CPU usage was the UI thread.

TLDR: It seems unmounting the Video Component on Android doesn't totally get rid of all processes.

Here's the example I have that took me about 30 min of running to run into this error, while hitting about 30% CPU usage in dev mode. https://github.com/xvalentino/react-native-video-leak/blob/master/index.android.js

**Comment by BeckZero** [comment]
+1 same problem...

**Comment by zchwyng** [comment]
Closing this since it's outdated.
